### PR TITLE
New parser logic.

### PR DESCRIPTION
**Pull request by rodrigovr**
This algorithm can deal with a pretty rare edge case of the sequence \n\n at the end of a event arriving on separated network packets.

### CODE REVIEW

**Comment by mpetazzoni** [comment]
Hi @rodrigovr,

Thanks for the PR! Does this fix #33 as well?

**Comment by rodrigovr** [comment]
hello @mpetazzoni !

This PR didn't originally fix #33 but I just added one more commit addressing the issue after reading the specs at https://html.spec.whatwg.org/multipage/server-sent-events.html#dispatchMessage

hope you can validade and accept it to the upstream

best regards

File: lib/sse.js [hunk 1]
```diff
@@ -163,16 +165,23 @@ var SSE = function (url, options) {
...
+        }
+        e['data'] = value;
       } else {
         e[field] = value;
       }
     }.bind(this));
 
     var event = new CustomEvent(e.event);
-    event.data = e.data;
+    event.data = e.data || '';
```
Comment (chahu) [inline lib/sse.js:184]: You may want to use ?? if data is 0

Comment (rodrigovr) [reply lib/sse.js:184]: ('0' || '') also resolves to '0'

**Pull request merged**